src/invidious/search/query.cr

```diff
@@ -48,10 +51,18 @@ module Invidious::Search
     )
       # Get the raw search query string (common to all search types). In
       # Regular search mode, also look for the `search_query` URL parameter
-      if @type.regular?
-        @raw_query = params["q"]? || params["search_query"]? || ""
-      else
-        @raw_query = params["q"]? || ""
+      _raw_query = params["q"]?
+      _raw_query ||= params["search_query"]? if @type.regular?
+      _raw_query ||= ""
+
+      # Remove surrounding whitespaces. Mostly useful for copy/pasted URLs.
+      @raw_query = _raw_query.strip
+
+      # Check for smart features (ex: URL search) inhibitor (exclamation mark).
+      # If inhibitor is present, remove it.
+      if @raw_query.starts_with?('!')
+        @inhibit_ssf = true
+        @raw_query = @raw_query[1..]
```

Comment:
Using `!` as an inhibitor seems a bit odd to me, especially when it functions as an escape. What do you think of using a backslash instead?

Reply:
I'm neutral on this

Reply:
If it doesn't make parsing more difficult then I'm in favor of using a backslash instead of an exclamation mark. Since usually an `!` is to _use_ a special smart functionality rather than negate it